extension.js

```diff
@@ -313,9 +317,50 @@ function workarea_for_window(win) {
 }
 
 function target_rect_for_workarea(workarea) {
-    const target_rect = workarea.copy();
-    target_rect.height *= settings.get_double('window-height');
-    return target_rect;
+    const available_width = workarea.width,
+        available_height = workarea.height,
+        full_width = available_width + workarea.x,
+        full_height =  available_height + workarea.y,
+        width = available_width * settings.get_double('window-width'),
+        height = available_height * settings.get_double('window-height'),
+        horizontal_alignment = settings.get_string('window-horizontal-alignment'),
+        vertical_alignment = settings.get_string('window-vertical-alignment');
+
+    // width = Math.min(full_width * settings.get_double('window-width'), available_width),
+    // height = Math.min(full_height * settings.get_double('window-height'), available_height),
+    let x, y;
+    switch (horizontal_alignment) {
+    case 'left':
+        x = workarea.x;
+        break;
+    case 'right':
+        x = full_width - width;
+        break;
+    case 'center':
+        var x_diff = available_width - width;
+        if (x_diff === 0)
+            x = workarea.x;
+        else
+            x = workarea.x + Math.round(x_diff / 2);
+        break;
+    }
+    switch (vertical_alignment) {
+    case 'top':
+        y = workarea.y;
+        break;
+    case 'bottom':
+        y = full_height - height;
+        break;
+    case 'center':
+        var y_diff = available_height - height;
+        if (y_diff === 0)
+            y = workarea.y;
+        else
+            y = workarea.y + Math.round(y_diff / 2);
```

Comment:
I think it should be possible to implement this simpler, without `if` / ternary operator

Reply:
Done like before.